src/commands/ask.ts

```diff
@@ -0,0 +1,118 @@
+import { ApplicationCommandOptionType, CommandInteraction, InteractionResponse, Message } from 'discord.js'
+import {
+  Discord,
+  SimpleCommand,
+  SimpleCommandMessage,
+  SimpleCommandOption,
+  SimpleCommandOptionType,
+  Slash,
+  SlashOption,
+} from 'discordx'
+
+const COMMAND_NAME = 'ask'
+const COMMAND_DESC = 'Ask a question to Wolfram Alpha'
+
+const COOLDOWN_MILLISECONDS = 3 * 60 * 1000
+
+@Discord()
+class Ask {
+  private apiToken: string
+  private lastUsage: number
+
+  constructor() {
+    this.apiToken = process.env.WOLFRAM_APP_ID ?? ''
+    this.lastUsage = 0
+
+    if (this.apiToken == '') {
+      throw Error('WOLFRAM_APP_ID needs to be set')
+    }
+  }
+
+  @SimpleCommand({ name: COMMAND_NAME, description: COMMAND_DESC })
+  async simple(
+    @SimpleCommandOption({ name: 'question', type: SimpleCommandOptionType.String })
+    question: string | undefined,
+    command: SimpleCommandMessage
+  ) {
+    if (!question) {
+      return await command.message.reply({ content: 'Usage: >ask <question>', allowedMentions: { repliedUser: false } })
+    }
+
+    if (this.isOnCooldown()) {
+      return
+    }
+
+    try {
+      const answer = await this.fetchAnswer(question)
+      return command.message.reply({ content: answer, allowedMentions: { repliedUser: false } })
+    } catch (err) {
+      console.error(err)
+      return command.message.reply({
+        content: 'There was a problem communicating with Wolfram Alpha.',
+        allowedMentions: { repliedUser: false },
+      })
+    }
+  }
+
+  @Slash({ name: COMMAND_NAME, description: COMMAND_DESC })
+  private async slash(
+    @SlashOption({
+      name: 'question',
+      type: ApplicationCommandOptionType.String,
+      description: 'The question you want to ask',
+      required: true,
+    })
+    question: string,
+    interaction: CommandInteraction
+  ): Promise<InteractionResponse<boolean>> {
+    const cooldownMessage = this.isOnCooldown()
+    if (cooldownMessage) {
+      return interaction.reply(cooldownMessage)
+    }
+
+    try {
+      const answer = await this.fetchAnswer(question)
+      return interaction.reply(`[${question}] ${answer}`)
+    } catch (err) {
+      console.error(err)
+      return interaction.reply('There was a problem communicating with Wolfram Alpha.')
+    }
+  }
+
+  private async fetchAnswer(question: string): Promise<string> {
+    const url = new URL('https://api.wolframalpha.com/v1/result')
+    url.searchParams.append('appid', this.apiToken)
+    url.searchParams.append('i', question)
+    url.searchParams.append('units', 'metric')
```

Comment:
Love it. The Americans might moan about freedom units though? Could we add an option for either?

Reply:
I added an option to use imperial measurements, using it on prefix commands is kinda jank though since I need to split the arguments by newline for the question, you need to specify the boolean in the next line like:
```
>ask How far is Berlin from London?
true
```

Reply:
Could it be selecting an option from `{'metric', 'imperial'}`? To make it a bit clearer than True/False? I can't remember how the discord options thing works.

Reply:
That would be better, you're right, I changed it now.